app/components/form_components/schools_autocomplete/script.js

```diff
@@ -0,0 +1,67 @@
+import accessibleAutocomplete from 'accessible-autocomplete'
+
+const $form = document.querySelector('[data-module="app-schools-autocomplete"]')
+const idElement = document.getElementById('lead-school-id')
+
+let statusMessage = null
+
+const findSchools = (query, populateResults) => {
+  statusMessage = 'Loading...'
+
+  window.fetch(`/api/schools?query=${query}&lead_school=true`)
+    .then(response => response.json())
+    .then(data => {
+      if (data === undefined) {
+        return
+      }
+
+      const schools = data.schools
+
+      if (schools.length === 0) {
+        statusMessage = 'No results found'
+      }
+
+      return schools
+    })
+    .then(schools => populateResults(schools))
+    .catch(err => console.log(err))
+}
+
+const suggestionTemplate = (value) => {
+  return value && value.name
+}
+
+const inputTemplate = (value) => {
+  return value && value.name
+}
+
+const renderTemplate = {
+  inputValue: inputTemplate,
+  suggestion: suggestionTemplate
+}
+
+const setParams = (value) => {
+  if (value === undefined) {
+    return
+  }
+  idElement.value = value.id
+}
```

Comment:
is there a need for early return?

```suggestion
const setLeadSchoolHiddenField = (school => {
  leadSchoolHiddenField.value = school && school.id
}
```